src/datumaro/components/merge/extractor_merger.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Optional, Sequence, TypeVar
+
+from datumaro.components.contexts.importer import _ImportFail
+from datumaro.components.dataset_base import DatasetBase, SubsetBase
+
+T = TypeVar("T")
+
+
+def check_identicalness(seq: Sequence[T], raise_error_on_empty: bool = True) -> Optional[T]:
+    if len(seq) == 0 and raise_error_on_empty:
+        raise _ImportFail("It should not be empty.")
+    elif len(seq) == 0 and not raise_error_on_empty:
+        return None
+
+    if seq.count(seq[0]) != len(seq):
```

Comment:
can this check identicalness really?

Reply:
Yes, `seq.count(seq[0])` count the same items with `seq[0]` from seq.